### PR TITLE
Add `Response#raw`

### PR DESCRIPTION
**Pull request by 0x1eef**
**Summary**

Fork of https://github.com/antaz/llm/pull/13

* Add `LLM::Response#raw` 
  The parsed response body from a LLM response
* Add `LLM::Response::Completion`
  Represents a response from a completion API.
  A subclass|type of `LLM::Response`. 
* Add `LLM::Completion#messages`
  Delegates to `@provider.completion_messages`
* Add `LLM::Provider#completion_messages`
  Returns an array of `LLM::Message` objects
  (After parsing a provider-specific data structure)
* Adjust require logic
  Our require logic was hard to follow and duplicated.
  Try to standardize a different approach

### CODE REVIEW

File: lib/llm/response/completion.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# frozen_string_literal: true
+
+module LLM
+  require_relative "../response"
```
Comment (0x1eef) [inline lib/llm/response/completion.rb:4]: This can be removed

File: lib/llm/providers/openai.rb [hunk 2]
```diff
@@ -33,7 +27,16 @@ def complete(prompt, params = {})
       auth req
       res = request @http, req
 
-      Completion.new(res.body, :openai)
+      Response::Completion.new(res.body, self)
+    end
+
+    ##
```
Comment (0x1eef) [inline lib/llm/providers/openai.rb:33]: We should add: `@api private`

File: lib/llm/providers/gemini.rb [hunk 3]
```diff
@@ -28,15 +23,23 @@ def complete(prompt, params = {})
...
-        JSON.parse(res.body)["candidates"].map { |candidate|
-          Message.new(candidate.dig("content", "role"), candidate.dig("content", "parts", 0, "text"))
-        }
-      )
+      Response::Completion.new(res.body, self)
     end
 
     private
 
+    ##
```
Comment (0x1eef) [inline lib/llm/providers/gemini.rb:31]: This was made private to hide it from IRB's auto-complete system.

**Comment by 0x1eef** [comment]
@antaz What do you think about this approach ? The `completion_messages` method has been made private as  suggested previously. That hides it from IRB autocomplete as well.

File: lib/llm/providers/gemini.rb [hunk 4]
```diff
@@ -28,15 +23,23 @@ def complete(prompt, params = {})
...
+    # @param (see LLM::Provider#completion_messages)
+    # @return (see LLM::Provider#completion_messages)
+    def completion_messages(raw)
+      raw["candidates"].map do
+        LLM::Message.new(
+          _1.dig("content", "role"),
+          _1.dig("content", "parts", 0, "text")
```
Comment (0x1eef) [inline lib/llm/providers/gemini.rb:38]: @antaz Should `Message#text` be an array ? In this context it can have "parts"

Comment (antaz) [reply lib/llm/providers/gemini.rb:38]: @0x1eef I'm not sure what Gimini refers to by parts, yet
